### PR TITLE
MET-4473 Support 3d tier calculation for sandbox

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java [hunk 1]
```diff
@@ -62,9 +62,11 @@ public final TierClassification<MediaTier, ContentTierBreakdown> classify(RdfWra
                                                           .orElse(MediaTier.T0);
       mediaResourceTechnicalMetadataList = descendingMediaResourceTechnicalMetadata;
     }
+    MediaType mediaTypeResult = getMediaType();
+    boolean hasMediaResource3DAvailable = mediaTypeResult == MediaType.THREE_D && (mediaTier != MediaTier.T0 && mediaTier != MediaTier.T1);
 
-    final ContentTierBreakdown contentTierBreakdown = new ContentTierBreakdown(getMediaType(), entityLicenseType, hasThumbnails,
-        hasLandingPage, hasEmbeddableMedia, mediaResourceTechnicalMetadataList);
+    final ContentTierBreakdown contentTierBreakdown = new ContentTierBreakdown(mediaTypeResult, entityLicenseType, hasThumbnails,
+        hasLandingPage, hasMediaResource3DAvailable,hasEmbeddableMedia, mediaResourceTechnicalMetadataList);
```
Comment (jeortizquan) [inline metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java:69]: We need to refactor this class ContentTierBreakdown the constructor is polluted with many boolean parameters. I think is not the scope of this story. Can you create a ticket to refactor this?

File: metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/ContentTierBreakdownTest.java [hunk 2]
```diff
@@ -67,4 +67,4 @@ void objectCreationTest_Without_Processing_Errors_Parameter() {
         contentTierBreakdown.getMediaResourceTechnicalMetadataList()));
 
   }
-}
\ No newline at end of file
+}
```
Comment (jeortizquan) [inline metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/ContentTierBreakdownTest.java:70]: A test is missing with ```true``` value to cover the new feature.

File: metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/RecordTierCalculationViewTest.java [hunk 3]
```diff
@@ -54,4 +54,4 @@ void objectCreationTest() {
     assertNotNull(recordTierCalculationView.getMetadataTierBreakdown());
   }
 
-}
\ No newline at end of file
+}
```
Comment (jeortizquan) [inline metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/RecordTierCalculationViewTest.java:57]: same here a test is missing with ```true``` to cover the new feature.

**Review by jeortizquan — COMMENTED** [review]
In my opinion needs refactor. See my comments, and check if they apply for this iteration.

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java (hunk 1 above)
Comment (JoanaCMS) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java]: I decided to handle this anyway and created a Builder.
This pollution should be gone now

File: metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/ContentTierBreakdownTest.java (hunk 2 above)
Comment (JoanaCMS) [reply metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/ContentTierBreakdownTest.java:80]: Fixed it

File: metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/RecordTierCalculationViewTest.java (hunk 3 above)
Comment (JoanaCMS) [reply metis-indexing/src/test/java/eu/europeana/indexing/tiers/view/RecordTierCalculationViewTest.java:64]: Fixed it

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java (hunk 1 above)
Comment (jeortizquan) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/AbstractMediaClassifier.java]: :+1:

**Pull request merged**